cpp/ql/src/semmle/code/cpp/MemberFunction.qll

```diff
@@ -162,7 +164,7 @@ class ConstMemberFunction extends MemberFunction {
  * };
  * ```
  */
-class Constructor extends MemberFunction {
+class Constructor extends MemberFunction, TaintFunction {
```

Comment:
This would be (I think) the first time we implement models outside of `semmle.code.cpp.models.implementations`. It means that the documentation will show `TaintFunction` as a superclass and that users will see autocomplete for the `hasTaintFlow` predicate, which takes arguments whose types are `Function{In,Out}put`. Those types were previously not exposed in `import cpp` either, but they get exposed by the non-private imports above.

Unless there are strong reasons to put the model implementations here, I'd much prefer to keep them in a new file in `semmle.code.cpp.models.implementations`.

Reply:
I was aware that this is the first time we implement a model outside of the 'models' directory, but not of the consequences you describe.  The reason I did so is that we already have classes which model _exactly_ the elements I'm interested in, so it seemed superfluous and mildly confusing to create new classes - but for the reasons you spell out it probably is worth doing exactly that.  It also keeps the models together in one place, which is nice.

Reply:
Done.